read-cache.c

```diff
@@ -791,7 +791,7 @@ int add_to_index(struct index_state *istate, const char *path, struct stat *st,
 int add_file_to_index(struct index_state *istate, const char *path, int flags)
 {
 	struct stat st;
-	if (lstat(path, &st))
+	if (lstat(path, &st) && !(flags & ADD_CACHE_IGNORE_ERRORS))
```

Comment:
Is this really a recoverable error? I would claim that it is an *unrecoverable* error, as a failed `lstat()` call leaves `st` uninitialized, yet it is used two lines further down. That would introduce a major bug.

Reply:
@qusielle Ping!